include/TextInput.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef TEXTINPUT_HPP_
+# define TEXTINPUT_HPP_
+
+# include "Wlroots.hpp"
+
+class InputMethodRelay;
+
+struct TextInputListeners
+{
+  struct wl_listener pending_focused_surface_destroy;
+};
+
+class TextInput : public TextInputListeners
+{
+public:
+  TextInput(InputMethodRelay& relay, struct wlr_text_input_v3 *wlr_text_input);
```

Comment:
```suggestion
  TextInput(InputMethodRelay& relay, struct wlr_text_input_v3 *wlr_text_input);
  TextInput(TextInput const &) = delete;
  TextInput(TextInput &&) = delete;
```

Reply:
This is actually not possible because i store unique_ptr of TextInput in a vector

Reply:
Should be possible if you didn't move the data into unique_ptr, going to check.